.github/workflows/build.yml

```diff
@@ -113,6 +114,55 @@ jobs:
           message: |
             Build succeeded and deployed at http://${{ env.surge_url }} 
             (hash ${{ github.event.pull_request.head.sha }} deployed at ${{ steps.date.outputs.date }})
+
+  master_frontend_build_and_deploy:
+    needs: frontend_tests
+    if: github.event_name == 'push' && github.ref == 'refs/heads/master'
+    name: master_frontend_build_and_deploy
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
```

Comment:
The inconvenient of a matrix is that it will run all the tasks times 3, including checkout and dependency building. Not a big deal but we could probably clean things up a bit. Could be a todo for someone else later if you don't see how right now.

Reply:
They are cached so it should be good

Reply:
From our discussion: we can benefit from the cache